pkg/api/common/const.go

```diff
@@ -186,37 +118,42 @@ func GetAllSupportedKubernetesVersions() []string {
 	return versions
 }
 
+// GetVersionsGt returns a list of versions greater than a semver string given a list of versions
+func GetVersionsGt(versions []string, version string) []string {
+	// Try to get latest version matching the release
+	var ret []string
+	for _, v := range versions {
+		sv, _ := semver.NewVersion(v)
+		cons, _ := semver.NewConstraint(">" + version)
+		if cons.Check(sv) {
+			ret = append(ret, v)
+		}
+	}
+	return ret
+}
+
 // AllKubernetesWindowsSupportedVersions maintain a set of available k8s Windows versions in acs-engine
-var AllKubernetesWindowsSupportedVersions = map[string]bool{
-	KubernetesVersion1Dot7Dot2:       true,
-	KubernetesVersion1Dot7Dot4:       true,
-	KubernetesVersion1Dot7Dot5:       true,
-	KubernetesVersion1Dot7Dot7:       true,
-	KubernetesVersion1Dot7Dot9:       true,
-	KubernetesVersion1Dot7Dot10:      true,
-	KubernetesVersion1Dot7Dot12:      true,
-	KubernetesVersion1Dot7Dot13:      true,
-	KubernetesVersion1Dot7Dot14:      true,
-	KubernetesVersion1Dot7Dot15:      true,
-	KubernetesVersion1Dot8Dot0:       true,
-	KubernetesVersion1Dot8Dot1:       true,
-	KubernetesVersion1Dot8Dot2:       true,
-	KubernetesVersion1Dot8Dot4:       true,
-	KubernetesVersion1Dot8Dot6:       true,
-	KubernetesVersion1Dot8Dot7:       true,
-	KubernetesVersion1Dot8Dot8:       true,
-	KubernetesVersion1Dot8Dot9:       true,
-	KubernetesVersion1Dot8Dot10:      true,
-	KubernetesVersion1Dot9Dot0:       true,
-	KubernetesVersion1Dot9Dot1:       true,
-	KubernetesVersion1Dot9Dot2:       true,
-	KubernetesVersion1Dot9Dot3:       true,
-	KubernetesVersion1Dot9Dot4:       true,
-	KubernetesVersion1Dot9Dot5:       true,
-	KubernetesVersion1Dot9Dot6:       true,
-	KubernetesVersion1Dot10Dot0Beta2: false,
-	KubernetesVersion1Dot10Dot0Beta4: false,
-	KubernetesVersion1Dot10Dot0RC1:   false,
+var AllKubernetesWindowsSupportedVersions = getAllKubernetesWindowsSupportedVersionsMap()
+
+func getAllKubernetesWindowsSupportedVersionsMap() map[string]bool {
```

Comment:
Instead of Windows support being a mostly duplicated whitelist of the larger support matrix, this is now an override/blackslist